Mammoth/TSE/CWeaponClass.cpp

```diff
@@ -2420,6 +2506,58 @@ bool CWeaponClass::FireWeapon (CInstalledDevice &Device,
 	if (Shots.GetCount() == 0)
 		return false;
 
+	//	If we're charging, then we don't fire shots - instead we only increment the ship counter (if needed),
+	//	create the fire effect, and return True (so that we consume power). Charging does not consume ammo,
+	//	and cannot fail due to damaged or disrupted weapons.
+	if (IsCharging)
+		{
```

Comment:
Probably should move this to a function and call it from here. Something like:

if (IsCharging)
   {
   return ChargeWeapon(...)
   }

That will keep FireWeapon cleaner